rfd/0174-kube-rbac-bootstrap.md

```diff
@@ -0,0 +1,346 @@
+We would create bindings for these roles on Cluster enrollment and allow using them through groups `default-view`/`default-edit`. This
+would allow users to always have an underlying role they could use when defining Teleport roles for Kubernetes access. This is a less complicated
+but also less powerful solution.
+
+We could also alternatively reverse targeting labels direction for Kube Provisioning. Instead of KubeProvision resource's labels
+selecting Kubernetes clusters for provisioning we could add a new field to the Kube service config, `kubeProvisionLabels`, 
+and these labels would select which resources should be provisioned to this cluster. E.g. if you don't have any `kubeProvisionLabels`
+setup on the Kube service, it won't "pull" any resources, which might be a bit more secure approach. But this also introduced more complexity
+in management, since the user can't easily change targeting just by creating/editing KubeProvision resource, they need to change config on every
+Kube service. This also looses granularity of control when doing discovery and proxying dynamic clusters, since all cluster will have same
+`kubeProvisionLabels` setup and user might need to run multiple Kube service to get the granularity back.
+
+## Future work
+
+As mentioned above adding pattern matching to the `kubernetes_permissions` field is probably something we will need to add in the future, improving
+fine-grained control of RBAC users can setup.
+
+Described scheme of auto-provisioning requires Kube service to have access to all Teleport roles, but in the future we plan migration to a more
+centralised approach to authorization, where edge services will not see the roles. When this happens we could replace the "ephemeral KubeProvision"
+approach with a real KubeProvision approach - for example some watcher on the Auth service could monitor roles and create/update actual KubeProvisions
+on the backend, designated with a special kind. And then Kube service will just use regular KubeProvision reconciliation to distribute desired RBAC
+to the target clusters.
+
+KubeProvision capabilities might be expanded to include any type of resource in the future, effectively acting something like a 
+Terraform for Kubernetes, but defined completely in Teleport. We should gather feedback after releasing initial KubeProvision
+feature to understand if there's demand for such extension of capabilities.
+
+Also in the future we can add a dashboard showing current state of KubeProvision reconciliation. We could add a `status` field to
+the KubeProvision resource definition, where we could track any errors that happened when KubeService tried to provision the resource and then 
+propagate that information to the UI.
+
+## Audit
+
+We will track changes to the KubeProvision resources in the Audit log. There will be three new events:
+
+* `kube.provision.create`
```

Comment:
As I mentioned earlier, ephemeral KubeProvision is just a logical abstraction and nothing actually is created. I don't think we need additional audit trail here, because it's already audited by the role manipulation - this auto-provisioning is directly linked to roles.